CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+if(ECMULT_GEN_PREC_BITS STREQUAL auto)
+  set(ECMULT_GEN_PREC_BITS 4)
+endif()
+if(NOT ECMULT_GEN_PREC_BITS STREQUAL 2 AND NOT ECMULT_GEN_PREC_BITS STREQUAL 4 AND NOT ECMULT_GEN_PREC_BITS STREQUAL 8)
+  message(FATAL_ERROR "ECMULT_GEN_PREC_BITS value is \"${ECMULT_GEN_PREC_BITS}\", but must an integer 2, 4, 8, or \"auto\".")
+endif()
+
+option(USE_FORCE_WIDEMUL_INT128 "force the use of the (unsigned) __int128 based wide multiplication implementation" OFF)
+option(USE_FORCE_WIDEMUL_INT64 "force the use of the (u)int64_t based wide multiplication implementation" OFF)
+if (USE_FORCE_WIDEMUL_INT128 AND USE_FORCE_WIDEMUL_INT64)
+  message(FATAL_ERROR "USE_FORCE_WIDEMUL_INT128 and USE_FORCE_WIDEMUL_INT64 cannot be enabled simultaneously.")
+endif()
+
+mark_as_advanced(FORCE
+  ENABLE_DEV_MODE
+  USE_FORCE_WIDEMUL_INT128
+  USE_FORCE_WIDEMUL_INT64
+)
+
+include(cmake/secp_try_append_cflag.cmake)
+secp_try_append_cflag(-pedantic)
+secp_try_append_cflag(-Wno-long-long)
+secp_try_append_cflag(-Wnested-externs)
+secp_try_append_cflag(-Wshadow)
+secp_try_append_cflag(-Wstrict-prototypes)
+secp_try_append_cflag(-Wundef)
+secp_try_append_cflag(-Wno-overlength-strings)
+secp_try_append_cflag(-Wall)
+secp_try_append_cflag(-Wno-unused-function)
+secp_try_append_cflag(-Wextra)
+secp_try_append_cflag(-Wcast-align)
+secp_try_append_cflag(-Wcast-align=strict)
+secp_try_append_cflag(-Wconditional-uninitialized)
+
+if(CMAKE_VERSION VERSION_GREATER 3.2)
+  # Honor visibility properties for all target types.
+  # See: https://cmake.org/cmake/help/latest/policy/CMP0063.html
+  cmake_policy(SET CMP0063 NEW)
+endif()
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+
+add_subdirectory(src)
+
+message("\n")
+message("Configure summary")
+message("=================")
+if(CMAKE_CROSSCOMPILING)
+  message("Cross compiling for ${CMAKE_SYSTEM_NAME}, ${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+message("Optional binaries:")
+message("  benchmark ........................ ${BUILD_BENCHMARK}")
+message("  tests ............................ ${BUILD_TESTS}")
+message("  exhaustive tests ................. ${BUILD_EXHAUSTIVE_TESTS}")
+message("  examples ......................... ${BUILD_EXAMPLES}")
+message("Optional modules:")
+message("  ECDH ............................. ${ENABLE_MODULE_ECDH}")
+message("  ECDSA pubkey recovery ............ ${ENABLE_MODULE_RECOVERY}")
+message("  extrakeys ........................ ${ENABLE_MODULE_EXTRAKEYS}")
+message("  schnorrsig ....................... ${ENABLE_MODULE_SCHNORRSIG}")
+message("Parameters:")
+message("  ecmult window size ............... ${ECMULT_WINDOW_SIZE}")
+message("  ecmult gen precision bits ........ ${ECMULT_GEN_PREC_BITS}")
+message("Optional features:")
+message("  external callbacks ............... ${USE_EXTERNAL_DEFAULT_CALLBACKS}")
+if(BUILD_TESTS OR BUILD_EXHAUSTIVE_TESTS)
+  message("  support kcov coverage analysis ... ${COVERAGE}")
```

Comment:
I just realized that this says "kcov" because the `configure --help` uses the same wording. I guess we should should update this when tidying up the configuration options as outlaid above. Then it would be helpful to reduce the doc strings in autotools and here to a minimum, and add larger explanations to the README. (This won't avoid duplication entirely but at least makes it less worse...)